src/app/products/charges/create-charge/create-charge.component.ts

```diff
@@ -0,0 +1,215 @@
+          return false;
+        }
+      }
+      return true;
+    });
+  }
+
+  /**
+   * Sets the conditional controls of the user form
+   */
+  setConditionalControls() {
+    this.chargeForm.get('chargeAppliesTo').valueChanges.subscribe((chargeAppliesTo) => {
+      switch (chargeAppliesTo) {
+        case 1: // Loan
+          this.chargeForm.addControl('chargePaymentMode', new FormControl('', Validators.required));
+          this.chargeForm.removeControl('incomeAccountId');
+          break;
+        case 2: // Savings
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          break;
+        case 3: // Client
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.addControl('incomeAccountId', new FormControl(''));
+          break;
+        case 4: // Shares
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          this.chargeForm.get('penalty').setValue(false);
+          break;
+      }
+      this.chargeForm.get('chargeCalculationType').reset();
+      this.chargeForm.get('chargeTimeType').reset();
+    });
+    this.chargeForm.get('chargeTimeType').valueChanges.subscribe((chargeTimeType) => {
+      this.chargeForm.removeControl('feeFrequency');
+      this.chargeForm.removeControl('feeInterval');
+      this.chargeForm.removeControl('feeOnMonthDay');
+      this.chargeForm.removeControl('addFeeFrequency');
+      switch (chargeTimeType) {
+        case 6: // Annual Fee
+          this.chargeForm.addControl('feeOnMonthDay', new FormControl('', Validators.required));
+          break;
+        case 7: // Monthly Fee
+          this.chargeForm.addControl('feeOnMonthDay', new FormControl(''));
+          this.chargeForm.addControl('feeInterval', new FormControl('', [Validators.required, Validators.min(1), Validators.max(12), Validators.pattern('^\\d*$')]));
+          this.repeatEveryLabel = 'Months';
+          break;
+        case 9: // Overdue Fee
+          this.chargeForm.get('penalty').setValue(true);
+          this.chargeForm.addControl('addFeeFrequency', new FormControl(false));
+          this.chargeForm.get('addFeeFrequency').valueChanges.subscribe((addFeeFrequency) => {
+            if (addFeeFrequency) {
+              this.chargeForm.addControl('feeFrequency', new FormControl('', Validators.required));
+              this.chargeForm.addControl('feeInterval', new FormControl('', [Validators.required, Validators.pattern('^\\d*$')]));
+            } else {
+              this.chargeForm.removeControl('feeFrequency');
+              this.chargeForm.removeControl('feeInterval');
+            }
+          });
+          break;
+        case 11: // Weekly Fee
+          this.chargeForm.addControl('feeInterval', new FormControl('', [Validators.required, Validators.pattern('^\\d*$')]));
```

Comment:
Remove the validator for pattern as I specified above.

Reply:
Did not remove as said above.